app/portainer/views/account/CreateAccessToken/CreateAccessToken.tsx

```diff
@@ -0,0 +1,107 @@
+import { PropsWithChildren, useState } from 'react';
+
+import { FormControl } from '@/portainer/components/form-components/FormControl';
+import { TextInput } from '@/portainer/components/form-components/Input';
+import { Button } from '@/portainer/components/Button';
+import { Heading } from '@/portainer/components/form-components/Heading';
+import { TextTip } from '@/portainer/components/Tip/TextTip';
+import { Code } from '@/portainer/components/Code';
+import { CopyButton } from '@/portainer/components/Button/CopyButton';
+
+import styles from './CreateAccessToken.module.css';
+
+interface AccessTokenResponse {
+  rawAPIKey: string;
+}
+
+export interface Props {
+  // userId for whom the access token is generated for
+  userId: number;
+
+  // onSubmit dispatches a successful matomo analytics event
+  onSubmit: (description: string) => Promise<AccessTokenResponse>;
+
+  // onSuccess is called when upon clicking the done button
+  onSuccess: () => void;
+
+  // onError is called when an error occurs; this is a callback to Notifications.error
+  onError: (heading: string, err: unknown, message: string) => void;
+}
+
+export function CreateAccessToken({
+  onSubmit,
+  onSuccess,
+  onError,
+}: PropsWithChildren<Props>) {
+  const [description, setDescription] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [accessToken, setAccessToken] = useState('');
+
+  // TODO: use axios or API service
```

Comment:
can remove or move to the service as it will be done in the service, no?

Reply:
actually, if you want to take some of the axios commit from`refactor/EE-1530/EE-1896/ng-services` feel free.

Reply:
and maybe it's a good place to try react-query?

Reply:
I haven't used react-query before.
Due to lack of time - i will not be doing it now.

The TODO is actually for before - since in a previous commit - was using raw fetch api. But then i moved the code to the caller (angularjs); ill remove the TODO.